### PR TITLE
DOP-2700: Organize related components

### PR DESCRIPTION
**Pull request by rayangler**
### Stories/Links:

DOP-2700

### Current Behavior:

See table below.

### Staging Links:

Staging links by component to ensure that components that were moved around still behave the same:

| Component | Current Behavior | Staging Link | Note |
| :--- | :--- | :--- | :--- |
| Breadcrumbs | [Atlas Prod](https://www.mongodb.com/docs/atlas/tutorial/create-atlas-account/) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/tutorial/create-atlas-account/)
| Footnote | [Atlas Prod](https://www.mongodb.com/docs/atlas/billing/subscriptions/) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/billing/subscriptions/) | See "[1]" on page |
| Contents | [Atlas Prod](https://www.mongodb.com/docs/atlas/billing/subscriptions/) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/billing/subscriptions/) | See "On This Page" in right column |
| Card | [Atlas Prod](https://www.mongodb.com/docs/atlas/#go-further-with-service) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/#go-further-with-service) |
| List | [Atlas Prod](https://www.mongodb.com/docs/atlas/security-vpc-peering/#in-enable-dns-hostnames-and-dns-resolution) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/security-vpc-peering/#in-enable-dns-hostnames-and-dns-resolution) | See list and list items "a" through "e" under the "AWS" tab |
| Tabs | [Atlas Prod](https://www.mongodb.com/docs/atlas/security-vpc-peering/#configure-an-service-network-peering-connection) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/security-vpc-peering/#configure-an-service-network-peering-connection)
| DefinitionList | [Atlas Prod](https://www.mongodb.com/docs/atlas/security/federated-authentication/) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/security/federated-authentication/) | See "Domain Mapping," "Organization Mapping", and "Role Mapping" after the first figure |
| Line | [Atlas Prod](https://www.mongodb.com/docs/atlas/triggers/cron-expressions/#field-values) | [Atlas Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/raymundrodriguez/DOP-2700/triggers/cron-expressions/#field-values) | See "All Values" and "(*)" in the table |
| Header | | | See any page's navbar + site banner |
| CSSClass | [Server Prod](https://www.mongodb.com/docs/manual/administration/production-checklist-development/#data-durability) | [Server Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/docs/raymundrodriguez/DOP-2700/administration/production-checklist-development/#data-durability) | See bulleted list under the first heading. This component is a little odd because its purpose is to apply custom class names (that we don't necessarily support, even prior to deletion of legacy css) |
| FieldList | [Server Prod](https://www.mongodb.com/docs/manual/reference/command/planCacheListFilters/#mongodb-dbcommand-dbcmd.planCacheListFilters) | [Server Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/docs/raymundrodriguez/DOP-2700/reference/command/planCacheListFilters/#mongodb-dbcommand-dbcmd.planCacheListFilters) | See "Returns" field after first table on the page |
| Figure | [Server Prod](https://www.mongodb.com/docs/manual/core/databases-and-collections/#collections) | [Server Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/docs/raymundrodriguez/DOP-2700/core/databases-and-collections/#collections) | 

Additional staging links for completion:

[Guides Staging](https://docs-mongodbcom-integration.corp.mongodb.com/test-guides/guides/raymundrodriguez/DOP-2700/index.html)
[Realm Staging](https://docs-mongodbcom-integration.corp.mongodb.com/master/realm/raymundrodriguez/DOP-2700/tutorial/ios-swift/index.html#part-1--set-up-the-mobile-app)

### Notes:
* Majority of the changes are moving components into subfolders and/or modifying the paths for imports.

### CODE REVIEW

**Review by gjchong25 — APPROVED** [review]
Thanks for organizing this, it looks much cleaner!! Just had one comment about naming convention consistency, but otherwise it looks great on the staging links (the Components table was super helpful btw!). Also this probably isn't under the scope of this ticket but I wonder if the CSSClass component is still necessary if we don't support the custom class names?

**Comment by rayangler** [comment]
> I wonder if the CSSClass component is still necessary if we don't support the custom class names?

@gjchong25 Yup, I agree that this component is questionable to keep around. I don't think we should remove it quite yet, especially since the parser still provides support for the directive in some capacity. We can investigate further and file a separate ticket to remove it as needed

**Comment by rayangler** [comment]
@gjchong25 I moved the remaining components to be in the form `/<Component>/index.js` to keep up with consistency. I also rebuilt the staging links above and included staging links for Guides and Realm to ensure that components built properly.

Re-requesting you as a reviewer just to make sure the new changes look good to you before merging!

File: src/components/DefinitionList/DefinitionList.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import ComponentFactory from './ComponentFactory';
+import ComponentFactory from '../ComponentFactory';
```
Comment (gjchong25) [inline src/components/DefinitionList/DefinitionList.js:4]: Should this file be `/DefinitionList/index.js` ?

Comment (rayangler) [reply src/components/DefinitionList/DefinitionList.js:4]: Yes, you're right. I missed `DefinitionList` and `FieldList`. I'll make the change!

**Pull request merged**